apps/dot_ext/authentication.py

```diff
@@ -0,0 +1,36 @@
+from django.contrib.auth.models import User
+from oauth2_provider.oauth2_validators import OAuth2Validator
+from rest_framework import authentication
+from rest_framework import exceptions
+from apps.fhir.authentication import extract_username
+
+
+class SLSAuthentication(authentication.BaseAuthentication):
+    def authenticate(self, request):
+        id = request.META.get('HTTP_X_AUTHENTICATION')
+        if not id:
+            return None
+
+        username = extract_username(id)
```

Comment:
I think this needs to check for username=None, or maybe those cases should throw an exception?

Reply:
Also, I'm not sure id is the best name for this variable

Reply:
Agreed on the `id`. The `username=None` case is covered by `test_unknown_authentication`, it ends up returning a 404.